store/sqlstore/oauth_store.go

```diff
@@ -69,7 +68,10 @@ func (as SqlOAuthStore) SaveApp(app *model.OAuthApp) (*model.OAuthApp, error) {
 		return nil, err
 	}
 
-	if err := as.GetMaster().Insert(app); err != nil {
+	if _, err := as.GetMasterX().NamedExec(`INSERT INTO OAuthApps
+		(Id, CreatorId, CreateAt, UpdateAt, ClientSecret, Name, Description, IconURL, CallbackUrls, Homepage, IsTrusted)
+		VALUES
+		(:Id, :CreatorId, :CreateAt, :UpdateAt, :ClientSecret, :Name, :Description, :IconURL, :CallbackUrls, :Homepage, :IsTrusted)`, app); err != nil {
```

Comment:
I can see this (and updates) becoming rather verbose for bigger structs. Is there any way we could generalize it a bit without becoming gorp again? :P

Reply:
Nope, I don't think so. The only alternative is to use reflection and therefore become gorp. :P

A slightly better way might be to use squirrel for big queries, just to keep it neat and tidy. I can suggest it more broadly when I demo it.

Reply:
I see. What if we generated a file with a big map of `[struct name]-->[db fields list]`? It's a bit more involved initially but it would avoid reflection and still support future schema alterations. Of course I am just brainstorming here, not asking for any immediate change.

Reply:
Definitely possible, but involves some manual work to keep them in sync. Personally, I am not a big fan of creating any abstractions over SQL queries. All of them eventually burn out in the long run, except maybe [sqlc](https://github.com/kyleconroy/sqlc), but it only supports MySQL for now.

I am a bit hesitant to start looking into solutions until we actually see this to be a problem. Writing raw SQL queries are just okay. Let's keep an eye on this for now.